docs_src/design/ucr/Bring-Your-Own-Vault.md

```diff
@@ -0,0 +1,75 @@
+
+If an adopter wishes to use an instance of Vault that they stand up or pre-exists in their environment, the EdgeX project does not provide any guidance or recipe for how to do this.  While technically possible, it would require a lot of work on the part of the adopter to:
+
+- Remove any EdgeX initialization / unsealing of the EdgeX Vault data store
+- Informs the existing Vault instance for EdgeX use
+- Configure the secrets engine of the existing Vault (allowing EdgeX permissions to initialize the Consul secrets engine).
+- Create policies, tokens, identities for all EdgeX services in the existing Vault
+- Inform EdgeX micro services to use the existing Vault instance
+- Populate existing Vault with EdgeX secrets
+
+In short, this is some tedious work and work that is not documented well (or in some cases at all).  It would require an adopter to study the secretstore-setup code and rework or replace the secretstore-setup service with new code to use the existing Vault instance.
+
+Therefore, the motivation for this EdgeX change is to make it easier to allow adopters to “bring their own Vault” instance and have EdgeX use that instance without any changes to the overall function of the EdgeX platform.
+
+## Target Users
+
+Any adopter that runs EdgeX in secure mode and with a pre-existing Vault or intention to share a Vault instance among edge applications.
+
+## Description
+
+Per the [issue](https://github.com/edgexfoundry/edgex-go/issues/1944) documented in the EdgeX GitHub edgex-go repository, the requirement is to:
+
+1. “de-privilege” EdgeX’s use of Vault
+2. Modularize security-secretstore-setup to allow portions to be replaced more easily with initialization / setup that uses an existing Vault instance
+3. Provide proper tools (scripts, code, etc.), documentation and examples to make it easier to replace EdgeX Vault instance with a 3rd party Vault instance.
+
+## Existing solutions
+
+One “hack” suggested by the EdgeX security working group chairman would be to fork the security secretstore setup so that it doesn't try to initialize the Vault instance and only performs a few of the tasks currently done by the secretstore setup code.  This assumes that Vault is running already with TLS in place and the existing Vault is using a non-expiring root token.
+
+The security of this alternative solution is very weak.
+
+## Requirements
+
+- Offer a “secure” EdgeX configuration deployment templates (via Docker Compose files, Snap/Snapcraft packaging, example Kubernetes Helm document, etc.) that uses a non-EdgeX established Vault instance (the Bring Your Own Vault or BYOV).
```

Comment:
Pick one or more of the deployment scenarios above and place it here.  We will need to test what we want to support.